server/jamovi/server/instancemodel.py

```diff
@@ -72,6 +74,28 @@ def __init__(self, instance):
 
         self.file_sync = None
 
+    @contextmanager
+    def attach(self):
+        ''' attach to the dataset for reading/writing'''
+        attach: Callable[[ ], None] | None
+        detach: Callable[[ ], None] | None
+
+        try:
+            attach = getattr(self._dataset, 'attach')
+            detach = getattr(self._dataset, 'detach')
+        except AttributeError:
+            attach = None
+            detach = None
+
+        if attach:
+            attach()
```

Comment:
Shouldn't this always happen?

Reply:
the shared memory dataset doesn't have an `attach()` method ... we can add one of course, but that will require a rebuild on macos, etc.